contract/contracts/selling-lock/src/entry.rs

```diff
@@ -1,39 +1,121 @@
+    debug!("lock.hash_type().as_reader().as_slice()[0] is {:?}",  lock.hash_type().as_reader().as_slice()[0]);
+    debug!("Byte::from(raw_args[32]).as_slice()[0] is {:?}",  Byte::from(raw_args[32]).as_slice()[0]);
+    debug!("lock.args().as_reader().raw_data()[..] is {:?}", lock.args().as_reader().raw_data());
+    debug!("raw_args[0..32] is {:?}", raw_args);
+    lock.code_hash().as_reader().raw_data()[..] == raw_args[0..32] &&
+    lock.hash_type().as_reader().as_slice()[0] == raw_args[32] &&
+    lock.args().as_reader().raw_data()[..] == raw_args[33..minimal_capacity_start_point]
+}
 
+fn check_is_owner(args: &Bytes) -> Result<bool, Error> {
+    let is_owner = QueryIter::new(load_cell_lock, Source::Input)
+        .find(|lock: &Script| {
+            debug!("check_is_owner:");
+            selling_lock_args_same_as_script(args, lock)
+        }).is_some();
+    Ok(is_owner)
+}
 
+fn get_self_capacity(args: &Bytes) -> u64 {
+    let mut buf = [0u8; MINIMAL_CAPACITY_LEN];
+    let self_cell = QueryIter::new(load_cell, Source::Input)
+        .find(|cell: &CellOutput| {
+            debug!("get_self_capacity:");
+            cell.as_reader().lock().args().raw_data()[..] == args.as_ref()[..]
+        }).unwrap();
+    buf.copy_from_slice(self_cell.as_reader().capacity().raw_data());
+    u64::from_le_bytes(buf)
+}
 
-    // return an error if args is invalid
-    if args.is_empty() {
-        return Err(Error::MyError);
-    }
+fn outputs_contains_owner_cell(args: &Bytes) -> Result<bool, Error> {
+    let contains_owner_cell = QueryIter::new(load_cell, Source::Output)
+        .find(|output: &CellOutput| {
+            debug!("outputs_contains_owner_cell:");
+            selling_lock_args_same_as_script(args, &output.lock())
+        }).is_some();
+    Ok(contains_owner_cell)
+}
+
+fn collect_outputs_owner_amount(args: &Bytes) -> Result<u64, Error> {
+    debug!("enter collect_outputs_owner_amount:");
+    let mut buf = [0u8; MINIMAL_CAPACITY_LEN];
+    let capacity_list = QueryIter::new(load_cell, Source::Output)
+        .map(|cell: CellOutput|{
+            debug!("now collect_outputs_owner_amount:");
+            debug!("selling_lock_args_same_as_script: {:?}",selling_lock_args_same_as_script(args, &cell.lock()));
+            debug!("cell.type_().as_reader().is_none(): {:?}",cell.type_().as_reader().is_none());
+            if selling_lock_args_same_as_script(args, &cell.lock()) && cell.type_().as_reader().is_none() {
+                debug!("&cell.capacity().raw_data(): {:?}",&cell.capacity().raw_data());
+                buf.copy_from_slice(&cell.capacity().raw_data());
+                return Ok(u64::from_le_bytes(buf));
+            } else {
+                return Ok(0u64)
+            }
+        }).collect::<Result<Vec<_>, Error>>()?;
+    Ok(capacity_list.into_iter().sum::<u64>())
+}
 
-    let tx_hash = load_tx_hash()?;
-    debug!("tx hash is {:?}", tx_hash);
+fn get_price(data: &Bytes) -> u64 {
```

Comment:
we should work with a group of scripts, image a scenario like this


```yml
- inputs:
  - owner_lock_a + price 1000 CKB
  - owner_lock_a + price 2000 CKB
  - owner_lock_a + price 3000 CKB
  - owner_lock_a + price 4000 CKB
  - owner_lock_a + price 5000 CKB
- outputs:
  - owner_lock_a + 5000 CKB
```

this unlock can be successful and will cause the owner's assets to be lost

Reply:
changed logic to iter over all input cells and get final price by the total sum, added one test case to cover this scenario